### ISSUE TITLE
[Feature] Captive Portal handling

### ISSUE THREAD

**Issue by jbicha** [budgie-desktop/budgie-desktop#807]
Feature Request
----------------------
After connecting to a network, NetworkManager (if enabled) checks whether the network is a captive portal. If it is, [GNOME Shell](https://help.gnome.org/misc/release-notes/3.14/#captive-portals) pops up the portals's webpage to allow the user to easily authenticate or whatever with the portal. This is powered by webkit2gtk. I think it would be useful if Budgie could do this too.

Distro Integration
----------------------
To enable the NetworkManager connectivity feature, you can create a file like
`/usr/lib/NetworkManager/conf.d/20-connectivity.conf` with contents like
```
[connectivity]
uri=http://nmcheck.gnome.org/check_network_status.txt
```
Fedora uses a separate package named [`config-connectivity-fedora`](http://pkgs.fedoraproject.org/cgit/rpms/NetworkManager.git/tree/NetworkManager.spec). A user can then easily disable the feature by uninstalling that package.

References
---------------
https://manpages.debian.org/unstable/network-manager/NetworkManager.conf.5.html#CONNECTIVITY_SECTION

**Comment by ikeydoherty** [comment]
Yeah I actually agree, this would've been so so handy last time I was in Portland. Damn hotel wifi wouldn't work properly ...
